icevision/parsers/via_parser.py

```diff
@@ -108,3 +115,38 @@ def bboxes(self, o) -> List[BBox]:
                         )
                     )
         return boxes
+
+
+class VIAMaskParser(VIABBoxParser, MasksMixin):
+    """
+    VIAMaskParser creates a mask from the `polygon` shape attribute
+    i.e. a collection of corresponding x-y coordinates
+    """
+
+    def masks(self, o) -> List[MaskArray]:
+        import PIL
+
+        masks = []
+        for shape in o["regions"]:
+            label = self._get_label(o, shape["region_attributes"])
+            if label in self.cls_map.class2id:
+                shape_attr = shape["shape_attributes"]
+                # assume that masks exist as polygons only
+                if shape_attr["name"] == "polygon":
+                    # create pairs of x-y points for plotting the polygon
+                    xy_coords = [
+                        (x, y)
+                        for x, y in zip(
+                            shape_attr["all_points_x"], shape_attr["all_points_y"]
+                        )
+                    ]
+
+                    # create empty image and fill it with the polygon
+                    # mask_img = PIL.Image.new("L", get_image_size(self.filepath(o)), 0)
+                    PIL.ImageDraw.Draw(mask_img).polygon(xy_coords, outline=1, fill=1)
```

Comment:
Instead of returning a image mask, can we directly return a `Polygon` class?

Reply:
The current `Polygon` class requires `points` to be vertices, like the COCO format. We'll need to convert `x,y` coordinate pairs to vertices to do that. I'm not sure how to do this, but I can look around for solutions